src/CMakeLists.txt

```diff
@@ -320,33 +322,34 @@ if (CMAKE_SYSTEM_NAME STREQUAL Linux)
     )
 
     add_custom_target(pal_redefines DEPENDS ${REDEFINES_FILE})
-
-    function(add_library_pal target_name do_redefinition lib_location)
-        get_filename_component(lib_file_name ${lib_location} NAME)
-        set(redefine_lib_location ${CMAKE_CURRENT_BINARY_DIR}/${lib_file_name})
-
-        if(${do_redefinition})
-            # Copy the PAL lib from it's original location and do the redefine
-            add_custom_command(
-                OUTPUT ${redefine_lib_location}
-                COMMAND ${OBJCOPY} --redefine-syms ${REDEFINES_FILE} ${lib_location} ${redefine_lib_location}
-                DEPENDS pal_redefines ${lib_location} ${REDEFINES_FILE}
-                COMMENT "Redefining exported PAL symbols in ${lib_location}"
-            )
-        else()
-            configure_file(${lib_location} ${redefine_lib_location} COPYONLY)
-        endif(${do_redefinition})
-
-        add_custom_target(${target_name}_redefines DEPENDS ${redefine_lib_location})
-
-        add_library(${target_name} STATIC IMPORTED)
-        set_property(TARGET ${target_name} PROPERTY IMPORTED_LOCATION ${redefine_lib_location})
-
-        add_dependencies(${target_name} ${target_name}_redefines)
-    endfunction()
 else()
-    message(FATAL_ERROR "Linux is only supported platform.")
```

Comment:
>message(FATAL_ERROR "Linux is only supported platform.") [](http://example.com/codeflow?start=4&length=56)

Rather than just allowing all builds to go through, could you do something like this:

elseif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
   message(NOTICE "Building for Darwin is for experimentation purposes")
else()
    message(FATAL_ERROR "CMake build not supported for system '${CMAKE_SYSTEM_NAME}'")
endif()